File: algorithms/strings/string-to-token/go/string_to_token.go
package stringtotoken

import "strings"

// tokenize splits text on a literal delimiter and omits empty tokens.
func tokenize(text, delimiter string) []string {
	if text == "" {
		return []string{}
	}
	if delimiter == "" {
		return []string{text}
	}

	parts := strings.Split(text, delimiter)
	result := make([]string, 0, len(parts))
	for _, part := range parts {
		if part != "" {
			result = append(result, part)
		}
	}

	return result
}

// Tokenize is an exported alias for tokenize.
func Tokenize(text, delimiter string) []string {
	return tokenize(text, delimiter)
}

File: algorithms/strings/string-to-token/java/StringToToken.java
import java.util.ArrayList;
import java.util.List;

public class StringToToken {
    public static String[] tokenize(String text, String delimiter) {
        if (text.isEmpty()) {
            return new String[0];
        }
        if (delimiter.isEmpty()) {
            return new String[]{text};
        }
        String[] parts = text.split(java.util.regex.Pattern.quote(delimiter), -1);
        List<String> result = new ArrayList<>();
        for (String part : parts) {
            if (!part.isEmpty()) {
                result.add(part);
            }
        }
        return result.toArray(new String[0]);
    }
}
